libcxx/include/__memory_resource/monotonic_buffer_resource.h

```diff
@@ -48,9 +47,11 @@ class _LIBCPP_AVAILABILITY_PMR _LIBCPP_EXPORTED_FROM_ABI monotonic_buffer_resour
       char* __end_;
       size_t __size_;
     };
-    void* __try_allocate_from_chunk(size_t, size_t);
   };
 
+  template <typename Chunk>
+  _LIBCPP_HIDE_FROM_ABI void* __try_allocate_from_chunk(Chunk& self, size_t bytes, size_t align);
```

Comment:
Can this be moved to a non-member function defined only in `memory_resource.cpp`?

Reply:
> Can this be moved to a non-member function defined only in `memory_resource.cpp`?

since struct __chunk_footer is private inside class monotonic_buffer_resource. then i couldn't get scope struct __chunk_footer __initial_descriptor.

but this can be easy archieved by add a branch judge cost.  otherwise i can distinguish them by struct size, but this is ugly.

Reply:
On the "ugliness": Oh, currently non-__ugllified identifiers are introduced in a header. If you want to add some implementation details to a header, the identifiers should be __uglified.

> but this can be easy archieved by add a branch judge cost.

I think the cost would be paid at only build-time - it should be just adding and manually specifying another template argument.

Reply:
done, thanks